pennylane/pauli/pauli_arithmetic.py

```diff
@@ -961,16 +961,16 @@ def _sum_same_structure_pws_dense(self, pauli_words, wire_order):
         ml_interface = qml.math.get_interface(coeff)
         if ml_interface == "torch":
             data0 = qml.math.convert_like(data0, coeff)
-        data = coeff * data0
+        data = coeff * data0 if np.isscalar(coeff) else np.outer(coeff, data0)
```

Comment:
Here we want to use `qml.math` instead of `np`. You can think of `qml.math` as the `np` of Tensorflow, Jax, Torch, Autograd and vanilla numpy all in one :) 
```suggestion
        data = coeff * data0 if qml.math.isscalar(coeff) else qml.math.outer(coeff, data0)
```